corehq/apps/hqadmin/management/commands/check_docs_in_es.py

```diff
@@ -0,0 +1,136 @@
+            extra_in_primary = stats.app_ids_primary - app_ids_es
+            _write_row('cases', extra_in_es, extra_in_primary)
+
+        if stats.num_users_es != stats.num_users_primary:
+            user_ids_es = set(UserES().domain(domain).get_ids())
+            extra_in_es = user_ids_es - stats.user_ids_primary
+            extra_in_primary = stats.user_ids_primary - user_ids_es
+            _write_row('users', extra_in_es, extra_in_primary)
+
+        if stats.num_groups_es != stats.num_groups_primary:
+            group_ids_es = set(GroupES().domain(domain).get_ids())
+            extra_in_es = group_ids_es - stats.group_ids_primary
+            extra_in_primary = stats.group_ids_primary - group_ids_es
+            _write_row('groups', extra_in_es, extra_in_primary)
+
+    def check_domains(self, all_domains, csvfile):
+        csvfile.writerow(['domain', 'doctype', 'docs_in_es', 'docs_in_primary_db'])
+        num_domains_es = DomainES().count()
+        num_domains_couch = Domain.get_db().view("domain/domains").all()[0]['value']
+
+        if num_domains_es != num_domains_couch:
+            # not a great start here
+            csvfile.writerow(["HQ", 'domains', num_domains_es, num_domains_couch])
+
+        domain_query = DomainES().fields(['name'])
+        if not all_domains:
+            domain_query = domain_query.is_active_project()
+        domains = domain_query.scroll()
+        for domain in domains:
+            self.check_domain(domain['name'], csvfile)
+
+    def check_domain(self, domain, csvfile):
+        stats = self.domain_info(domain)
+
+        if stats.num_cases_es != stats.num_cases_primary:
+            csvfile.writerow([domain, 'cases', stats.num_cases_es, stats.num_cases_primary])
+        if stats.num_forms_es != stats.num_forms_primary:
+            csvfile.writerow([domain, 'forms', stats.num_forms_es, stats.num_forms_primary])
+        if stats.num_apps_es != stats.num_apps_primary:
+            csvfile.writerow([domain, 'apps', stats.num_apps_es, stats.num_apps_primary])
+        if stats.num_users_es != stats.num_users_primary:
+            csvfile.writerow([domain, 'users', stats.num_users_es, stats.num_users_primary])
+        if stats.num_groups_es != stats.num_groups_primary:
+            csvfile.writerow([domain, 'groups', stats.num_groups_es, stats.num_groups_primary])
+
+    def domain_info(self, domain):
+        DomainStats = namedtuple('DomainStats', [
+            'num_cases_es', 'num_forms_es', 'num_apps_es', 'num_users_es', 'num_groups_es',
+            'num_cases_primary', 'num_forms_primary', 'num_apps_primary', 'num_users_primary', 'num_groups_primary',
+            'app_ids_primary', 'user_ids_primary', 'group_ids_primary', 'case_ids_primary', 'form_ids_primary'
+        ])
+
+        case_ids_primary = set(CaseAccessors(domain).get_case_ids_in_domain())
+        form_ids_primary = set(FormAccessors(domain).get_all_form_ids_in_domain())
```

Comment:
Yeah. I'm running into some issues with big domains https://github.com/dimagi/commcare-hq/pull/14259/commits/68e461bf9c8ffe3ec00c53d97b072ffbcc9d6d62 :/ I don't think it's OOM, but it's not a clear error